spec.html

```diff
@@ -3537,6 +3548,13 @@ <h1>The List and Record Specification Types</h1>
       <p>Schema for commonly used Record field combinations may be named, and that name may be used as a prefix to a literal Record value to identify the specific kind of aggregations that is being described. For example: PropertyDescriptor { [[Value]]: 42, [[Writable]]: *false*, [[Configurable]]: *true* }.</p>
     </emu-clause>
 
+    <emu-clause id="sec-map-specification-type">
+      <h1>The Map Specification Types</h1>
+      <p>The <dfn>Map</dfn> type is used to explain the evaluation of key-value based data structures, such as the metadata data structure built up by decorators (see <emu-xref href="#sec-metadata-record-specification-type"></emu-xref>). Values of the Map type are ordered sequences of key/value pairs containing individual values. These sequences may be of any length. The elements of a map may be randomly accessed using key-based indices. For notational convenience a dynamic-key syntax can be used to access Map elements. For example, _publicMetadata_[*"key"*] is shorthand for saying the value associated with *"key"* in the Map _publicMetadata_. If the map does not have a value associated with that key, then it returns ~empty~.</p>
+
+      <p>Maps can be iterated like lists, and are iterated as key/value pairs. If no order is specified, then elements are iterated in the order they were inserted.</p>
+    </emu-clause>
```

Comment:
why do we need this type at all? couldn't we make a null object (one that's never observed to exist) and use that for the lookups?